src/sentry/integrations/slack/client.py

```diff
@@ -1,22 +1,49 @@
 from typing import Any, Mapping, Optional, Union
 
-from requests import Response
+from requests import PreparedRequest, Response
 from sentry_sdk.tracing import Transaction
 
-from sentry.integrations.client import ApiClient
+from sentry.constants import ObjectStatus
+from sentry.models.integrations.integration import Integration
+from sentry.services.hybrid_cloud.util import control_silo_function
 from sentry.shared_integrations.client import BaseApiResponse
+from sentry.shared_integrations.client.proxy import IntegrationProxyClient
 from sentry.shared_integrations.exceptions import ApiError
+from sentry.types.integrations import EXTERNAL_PROVIDERS, ExternalProviders
 from sentry.utils import metrics
 
 SLACK_DATADOG_METRIC = "integrations.slack.http_response"
 
 
-class SlackClient(ApiClient):  # type: ignore
+class SlackClient(IntegrationProxyClient):
     allow_redirects = False
     integration_name = "slack"
     base_url = "https://slack.com/api"
     metrics_prefix = "integrations.slack"
 
+    @control_silo_function
+    def authorize_request(self, prepared_request: PreparedRequest) -> PreparedRequest:
+        integration = None
+        base_qs = {
+            "provider": EXTERNAL_PROVIDERS[ExternalProviders.SLACK],
+            "status": ObjectStatus.ACTIVE,
+        }
+        if self.integration_id:
+            integration = Integration.objects.filter(id=self.integration_id, **base_qs).first()
+        elif self.org_integration_id:
+            integration = Integration.objects.filter(
+                organizationintegration__id=self.org_integration_id, **base_qs
+            ).first()
+
+        if not integration:
+            return prepared_request
+
+        token = (
+            integration.metadata.get("user_access_token") or integration.metadata["access_token"]
+        )
+        prepared_request.headers["Authorization"] = f"Bearer {token}"
```

Comment:
The request in `src/sentry/integrations/slack/actions/notification.py` had the token in the payload. Does slack accept tokens in the payload or as a request header?

Reply:
Yeah, I was looking into this and it seems that slack [accepts the token](https://api.slack.com/methods/chat.postMessage#arg_token) in the payload or the header without a preference for either. It's not related to the `user_access_token` vs `access_token` which was my first guess.

 For logging purposes I think it's safer to keep the token in the header going forward.